src/SimpleCalculator.Web/Controllers/HomeController.cs

```diff
@@ -65,6 +72,17 @@ public IActionResult Privacy()
         return View();
     }
 
+    public IActionResult CalculationResults(int? page)
+    {
+        var pageNumber = page ?? 1;
+        int pageSize = 6;
+
+        var calcResultPageEntities = _context.CalculationResultEntities.ToList();
```

Comment:
basically you just get all data from DB, convert it to `CalculationEntityModel` model and only after that you apply pagination. Please consider a case where you apply pagination in the beginning and get only what you need.